lib/compress.js

```diff
@@ -3112,6 +3056,63 @@ merge(Compressor.prototype, {
                 }
             }
         }
+        if (compressor.option("unsafe_Func")
+            && exp instanceof AST_SymbolRef
+            && exp.undeclared()
+            && exp.name == "Function") {
+            // new Function() => function(){}
+            if (self.args.length == 0) return make_node(AST_Function, self, {
+                argnames: [],
+                body: []
+            });
+            if (all(self.args, function(x) {
+                return x instanceof AST_String;
+            })) {
+                // quite a corner-case, but we can handle it:
+                //   https://github.com/mishoo/UglifyJS2/issues/203
+                // if the code argument is a constant, then we can minify it.
+                try {
+                    var code = "NaN(function(" + self.args.slice(0, -1).map(function(arg) {
```

Comment:
Oh, I see. It's a regression in `2.8.x` and `3.x` caused by IIFE optimizations in recent months.
```
$ uglifyjs -V
uglify-js 2.7.5

$ echo 'Function("foo", "bar", "return foo + bar")' | uglifyjs -c unsafe
Function("a","b","return a+b");
```
Without this PR:
```
$ bin/uglifyjs -V
uglify-js 3.0.13

$ echo 'Function("foo", "bar", "return foo + bar")' | bin/uglifyjs -c unsafe
Function("foo","bar","return foo + bar");

$ echo 'Function("foo", "bar", "return foo + bar")' | bin/uglifyjs -c unsafe,side_effects=0
Function("a","b","return a+b");
```